test/functional/wallet_balance.py

```diff
@@ -77,8 +77,18 @@ def run_test(self):
         self.log.info("Mining blocks ...")
         self.generate(self.nodes[0], 1)
         self.generate(self.nodes[1], 1)
+
+        # Verify listunspent returns immature coinbase if 'include_immature_coinbase' is set
+        assert_equal(len(self.nodes[0].listunspent(query_options={'include_immature_coinbase': True})), 1)
+        assert_equal(len(self.nodes[0].listunspent(query_options={'include_immature_coinbase': False})), 0)
+
         self.generatetoaddress(self.nodes[1], COINBASE_MATURITY + 1, ADDRESS_WATCHONLY)
 
+        # Verify listunspent returns all immature coinbases if 'include_immature_coinbase' is set
+        # For now, only the legacy wallet will see the coinbases going to the imported 'ADDRESS_WATCHONLY'
+        assert_equal(len(self.nodes[0].listunspent(query_options={'include_immature_coinbase': False})), 1 if self.options.descriptors else 2)
+        assert_equal(len(self.nodes[0].listunspent(query_options={'include_immature_coinbase': True})), 1 if self.options.descriptors else COINBASE_MATURITY + 2)
```

Comment:
tiny tiny nit: the `COINBASE_MATURITY + 2` left me wondering for a sec, this might be easier to read if we had a variable storing how many blocks were generated:
```
n_blocks_generate = COINBASE_MATURITY + 2  # feel free to use a better name :)
self.generatetoaddress(self.nodes[1],  n_blocks_generate, ADDRESS_WATCHONLY)
...
assert_equal(len(self.nodes[0].listunspent(query_options={'include_immature_coinbase': True})), 1 if self.options.descriptors else n_blocks_generate + 1)
```

Reply:
k, as it is a tiny tiny nit, will only do it if have to retouch the code so we don't loose the ACKs.